app/controllers/courses_controller.rb

```diff
@@ -197,6 +197,9 @@ def notify_untrained
   def delete_all_weeks
     @course = find_course_by_slug(params[:id])
     @course.weeks.destroy_all
+    if @course.approved? 
+      DeletedTimelineAlertManager.new(course)
```

Comment:
I think this needs a better name than `DeletedTimelineAlertManager`, one that more clearly communicates that it works like a service object (it does a single task). Maybe something like `CheckTimeline` would be a good name, as that's the core idea, to check the timeline after it's been modified and see if anything needs to be done (eg, create an alert).